### ISSUE TITLE
FastLSTM does not use peepholes? -> README

### ISSUE THREAD

**Issue by kmnns** [Element-Research/rnn#82]
Thanks a lot for this extremely flexible toolkit!

I experience a pretty heavy performance drop when switching from `LSTM` to `FastLSTM`.

`FastLSTM` seems to discard cell2gate communication (i.e. no peepholes) and can thus mush ingate, forgetgate, outgate, and blockinput into one big structure (see below).

Is this true? If so it may be a good idea to mention that briefly in the [README](https://github.com/Element-Research/rnn#rnn.FastLSTM). Maybe also with other assumptions made for this class, if there are any.

```
nn.Sequential {
  [input -> (1) -> (2) -> (3) -> (4) -> (5) -> output]
  (1): nn.ConcatTable {
    input
      |`-> (1): nn.Sequential {
      |      [input -> (1) -> (2) -> (3) -> (4) -> (5) -> (6) -> output]
      |      (1): nn.NarrowTable
      |      (2): nn.ParallelTable {
      |        input
      |          |`-> (1): nn.Linear(13 -> 32)
      |          |`-> (2): nn.LinearNoBias(8 -> 32)
      |           ... -> output
      |      }
      |      (3): nn.CAddTable
      |      (4): nn.Reshape(4x8)
      |      (5): nn.SplitTable
      |      (6): nn.ParallelTable {
      |        input
      |          |`-> (1): nn.Sigmoid
      |          |`-> (2): nn.Tanh
      |          |`-> (3): nn.Sigmoid
      |          |`-> (4): nn.Sigmoid
      |           ... -> output
      |      }
      |    }
      |`-> (2): nn.SelectTable
       ... -> output
  }
```

**Comment by nicholas-leonard** [comment]
@kmnns The FastLSTM functionally the same as LSTM (see this [unit test](https://github.com/Element-Research/rnn/blob/master/test/test.lua#L397)). The difference is one of speed where the FastLSTM requires only two Linear calls *in total* as opposed to LSTM requiring two Linear calls *for each gate*. Both have peephole connections.

**Comment by kmnns** [comment]
@nicholas-leonard  Thank you for your answer. Still hunting down the cause for the performance drop with the exact same surrounding architecture. The unit test initializes the comparing LSTM [here](https://github.com/Element-Research/rnn/blob/master/test/test.lua#L410) as such:

```lua
local lstm1 = nn.LSTM(inputSize, inputSize, nil, false)
```

The final `false` disables the peepholes [here](https://github.com/Element-Research/rnn/blob/master/LSTM.lua#L19), doesn't it?

```lua
self.cell2gate = (cell2gate == nil) and true or cell2gate
```

**Comment by nicholas-leonard** [comment]
@kmnns Yes, you are right! Could you submit a PR updating the doc to mention this?

**Comment by kmnns** [comment]
Alright will do!

**Comment by nicholas-leonard** [comment]
fixed by #85
